src/common/mfu_daos.c

```diff
@@ -1313,3 +1317,102 @@ int mfu_flist_copy_daos(
 
     return rc;
 }
+
+int mfu_daos_hdf5_copy(mfu_file_t *mfu_src_file,
+                       mfu_file_t *mfu_dst_file,
+                       char **argpaths,
+                       daos_args_t *daos_args)
+{
+    int rc = 0;
+    int size;
+
+    MPI_Comm_size(MPI_COMM_WORLD, &size);
+    if (size > 1) {
+        MFU_LOG(MFU_LOG_ERR, "dcp uses h5repack to copy DAOS HDF5 containers "
+                "to and from a POSIX filesystem. Only one process "
+                "is currently supported, restart program using only "
+                "one MPI process "
+                MFU_ERRF, MFU_ERRP(-MFU_ERR_INVAL_ARG));
+                rc = 1;
+                return rc;
+    }
+
+    if (mfu_src_file->type == HDF5 || mfu_dst_file->type == HDF5) {
+        pid_t child;
+        int child_status;
+        pid_t child_pid;
+        char *args[4];
+        args[0] = "h5repack";
+
+        /* TODO: since the daos:// interface does not work yet in
+         * h5repack, this has not been properly tested. The DAOS->DAOS
+         * case is also currently not working, and needs to be tested */
+
+        /* cleanup pools/conts since h5repack will open/connect on its own */
+        daos_cleanup(daos_args, mfu_src_file, mfu_dst_file);
+
+        if (mfu_src_file->type == POSIX) {
+            args[1] = "--src-vol-name=native";
+            args[2] = strdup(argpaths[0]);
+            args[3] = strdup(argpaths[1]);
+            args[4] = NULL; 
+        } else if (mfu_dst_file->type == POSIX) {
+            /* if the src file type is HDF5, then we cannot
+             * assume that movement is going from DAOS->POSIX,
+             * since it may be from DAOS->DAOS. */
+            args[1] = "--dst-vol-name=native";
+            args[2] = strdup(argpaths[0]);
+            args[3] = strdup(argpaths[1]);
+            args[4] = NULL; 
+        } else {
+            /* this must be a DAOS->DAOS copy */
+            args[1] = "";
+            args[2] = strdup(argpaths[0]);
+            args[3] = strdup(argpaths[1]);
+            args[4] = NULL;
+        }
+
+        /* Child process */
+        if ((child = fork()) == 0) {
```

Comment:
Just a heads up that some MPI libraries do a poor job supporting fork() calls.  That means some users may encounter problems if their MPI library or their system happens to be one that has trouble with fork.  On the first IBM BlueGene systems, fork() wasn't an option at all.  On Infiniband networks, users may need to ``export IBV_FORK_SAFE=1`` to get things to work properly.

I don't know of a good substitute that can be used instead.  Just something to be aware of.  Perhaps add a note in our documentation where it would come into play.

Reply:
Oh, if HDF offers an API/library for this, that might be better than invoking their binary via fork().

Reply:
This is good to know. Whatever we ultimately end up doing, I'll be sure to run tests with the different systems and configurations that DAOS supports.